### ISSUE TITLE
Remove catches for non-Trick exceptions

### ISSUE THREAD

**Issue by alexlin0** [nasa/trick#220]
We currently catch all exceptions and call exit if an std::exception or an unknown exception is thrown.  This masks the source of the exception when we catch them.  The easier way to debug these problems is to not catch these exceptions and let the process core dump with an unhandled exception.  The core dump will show the origin of the exception.

**Comment by DavidHammen** [comment]
Hi, Alex!

If you've stepped up to c++11, you can use [`std::throw_with_nested`](http://en.cppreference.com/w/cpp/error/throw_with_nested) to capture the context without masking it, and then [`std::rethrow_if_nested`](http://en.cppreference.com/w/cpp/error/rethrow_if_nested) to print the nested error messages.

My experience is that printed messages are much nicer than core dumps when the problem results from a user error, but it's the other way around when the problem results from a programmer error. So use `assert` to test for probably programmer errors. My latest project is chock-full of assertions to test for my errors, throws to deal with user errors, and throw_with_nested to add context to those thrown exceptions.

**Comment by alexlin0** [comment]
This issue came up because we have an unknown exception like a  `throw 42` where we have found the best information to locate the source of the exception is letting the sim core dump.

I'll take a look at the nested throws for the standard exceptions.

**Comment by alexlin0** [comment]
I experimented with throwing nested and rethrowing exceptions, but in all cases I found that once I did a "catch" I lost the stack information I need to pinpoint the error.  So I went a different route, I don't catch std::exceptions or unknown exceptions.  I let std::terminate get called to handle these.  std::terminate calls abort which triggers a SIGABRT.  I put a signal handler in for SIGABRT that prints uses the same handler we have for SIGSEGV which prints out a stack trace right before exiting the process.  This stack trace contains the throw's original location.
